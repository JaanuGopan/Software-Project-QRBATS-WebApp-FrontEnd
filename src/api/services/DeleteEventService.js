import axios from "axios";
import ApiConstants from "../config/ApiConstants";

class DeleteEventService {
  static async deleteEvent(eventId) {
    try {
      const response = await axios.post(ApiConstants.deleteEventUrl, {
        eventId: eventId,
      });
      console.log("deleted Successfully..");
      return response.data.token;
    } catch (error) {
      throw new Error("Delete Failed ", error);
    }
  }
}

export default DeleteEventService;
