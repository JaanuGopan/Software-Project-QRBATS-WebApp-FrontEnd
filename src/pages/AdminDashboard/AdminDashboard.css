@media screen and (max-width: 1800px){
  .mainHead {
    font-size: x-large;
    margin-top: 5px;
    padding-left: 35px;
  }
  .informationText{
    font-size: large;
  }
  .buttonIcon{
    height: 1.5rem;
    width: 1.5rem;
    size: 1.5rem;
  }
  .NormalButton{
    height: 2.5rem;
    padding: 0.2vw;
    padding-left: 1vw;
  }
  .NormalButtonLabel{
    font-size: larger;
    padding: 1vw;
    margin-bottom: 0px;
  }
  .admin-Dash{
    margin: 0px;
  }
  .SearchEvent{
    margin-right: 35px;
  }
}

@media screen and (max-width: 768px){
  .mainHead {
    font-size: large;
    margin-top: 10px;
    padding-left: 35px;
  }
  .informationText{
    font-size: large;
  }
  .buttonIcon{
    height: 2rem;
    width: 2rem;
    size: 2rem;

  }
  .NormalButtonLabel{
    font-size: small;
    margin-bottom: 0px;
  }
  .admin-Dash{
    margin: 0px;
  }
  .SearchEvent{
    margin-right: 35px;
  }
}

@media screen and (max-width: 426px){
  .informationText{
    font-size: medium;
  }
  .NormalButtonLabel{
    font-size: small;
    margin-bottom: 0px;
  }
  .buttonIcon{
    height: 1rem;
    width: 1rem;
    size: 1rem;
  }
  .admin-Dash{
    margin: 0px;
  }
  .SearchEvent{
    margin-right: 0px;
  }
  .mainHead {
    padding-left: 5px;
  }
  .buttonIcon{
    height: 1.5rem;
    width: 1.5rem;
    size: 1.5rem;
  }
}

.admin-Dash{
  flex-grow: 1; 
  height: 90vh;
  width: 100%;
}

.mainHead {
  color: #0D295E;
  font-weight: bold;
  margin-bottom: 0px;
}

.mainInform {
  display: flex;
  flex-direction: row;
  align-items: center;
  flex-wrap: wrap;
}

.inform {
  display: flex;
  flex: 1;
  flex-direction: column;
  align-items: stretch;
  border: 3px solid white;
  border-radius: 10px;
  padding: 2%;
  width: 20vw;
  height: 20vh;
  margin: 1%;
  min-width: 115px;
  background: linear-gradient(#0063A0, rgba(0,100,160,0.8));
}

.inform1 {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}

.AdminEventList{
  display: flex;
  flex-direction: column;
  margin-top: 10px;
  padding-left: 10px;
  padding-right: 10px;
  padding: 10px;
  width: 100%;
  height: 55vh;
  border-top: 5px solid #0365A1;
  box-shadow: 0.5px 2px 0.5px 0.5px rgba(25, 65, 85, 0.5);
  border-radius: 10px;
  background-color: #FAFAFA;
}

.tableDesign {
 /* position: relative;
  height: 350px;
  display: block;*/
  width: 100%;
  height: -webkit-fill-available;
}

.tableArrangement {
  display: block;
  overflow: hidden;
  table-layout: auto;
  border-collapse: collapse;
  box-shadow: 0px 5px 5px #ccc;
  border-radius: 10px;
  white-space: nowrap;
  width: 100em;
  max-width: 90%;
  margin: auto;
  overflow-x: auto;
  height: 45vh;
  overflow-y: auto;
  border-bottom: 2px solid #0063A0;
}

.student-tableArrangement{
  display: block;
  overflow: hidden;
  table-layout: auto;
  border-collapse: collapse;
  box-shadow: 0px 5px 5px #ccc;
  border-radius: 10px;
  white-space: nowrap;
  width: 100em;
  max-width: 90%;
  margin: auto;
  overflow-x: auto;
  height: 65vh;
  overflow-y: auto;
  border-bottom: 2px solid #0063A0;
}

.event-report-tableArrangement{
  display: block;
  overflow: hidden;
  table-layout: auto;
  border-collapse: collapse;
  box-shadow: 0px 5px 5px #ccc;
  border-radius: 10px;
  white-space: nowrap;
  width: 100em;
  max-width: 90%;
  margin: auto;
  overflow-x: auto;
  height: 75vh;
  overflow-y: auto;
  border-bottom: 2px solid #0063A0;
}

.tableArrangement thead{
  position: sticky;
  top: 0;
  background-color: #c1e2f9;
  color: #0D295E;
  border-bottom: 3px solid #0063A0;
  text-align: center;
}

.student-tableArrangement thead{
  position: sticky;
  top: 0;
  background-color: #c1e2f9;
  color: #0D295E;
  border-bottom: 3px solid #0063A0;
  text-align: center;
}

.event-report-tableArrangement thead{
  position: sticky;
  top: 0;
  background-color: #c1e2f9;
  color: #0D295E;
  border-bottom: 3px solid #0063A0;
  text-align: center;
}

.tableArrangement th{
  padding: 0.8rem;
  text-align: center;
}

.student-tableArrangement th{
  padding: 0.8rem;
  text-align: center;
}

.event-report-tableArrangement th{
  padding: 0.8rem;
  text-align: center;
}

.tableArrangement td{
  overflow: hidden;
  text-overflow: ellipsis;
  text-align: center;
  padding: 0.5rem;
}

.student-tableArrangement td{
  overflow: hidden;
  text-overflow: ellipsis;
  text-align: center;
  padding: 0.5rem;
}

.event-report-tableArrangement td{
  overflow: hidden;
  text-overflow: ellipsis;
  text-align: center;
  padding: 0.5rem;
}

.tableArrangement tbody{
  overflow-y: auto;
}

.student-tableArrangement tbody{
  overflow-y: auto;
}

.event-report-tableArrangement tbody{
  overflow-y: auto;
}

.tableArrangement tbody tr{
  border: 0.01px solid rgba(91, 178, 255, 0.608);
  border-radius: 5px;
}

.student-tableArrangement tbody tr{
  border: 0.01px solid rgba(91, 178, 255, 0.608);
  border-radius: 5px;
}

.event-report-tableArrangement tbody tr{
  border: 0.01px solid rgba(91, 178, 255, 0.608);
  border-radius: 5px;
}

.tableArrangement tbody tr:hover{
  background-color: azure;
  cursor: pointer;
}

.student-tableArrangement tbody tr:hover{
  background-color: azure;
  cursor: pointer;
}

.event-report-tableArrangement tbody tr:hover{
  background-color: azure;
  cursor: pointer;
}

.expand {
  width: 100%;
}

.List-Buttons{
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  align-items: center;
  justify-content: space-around;
  padding: 1rem;
}

.NormalButton{
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  align-items: center;
  justify-content: center;
  color: white;
  cursor: pointer;
  background-color: #0063A0;
  border-radius: 10px;
}

.NormalButton:hover{
  border: 1px solid #025587;
  background-color: #2273a5;
}

.EditButton{
  border-radius: 30px;
  background-color: #0063A0;
  color: white;
  padding: 10px 15px;
}

.EditButton:hover{
  background-color: blue;
}

.Admin-Create-Event-Dashboard{
  position: fixed;
  z-index: 2;
  left: 0;
  top: 0;
  overflow-y: scroll;
  width: 100%;
  height: 100vh;
  padding-top: 5vh;
  padding-bottom: 5vh;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: rgba(0, 0, 0, 0.5);
}

.EditIcon{
  padding: 0px;
}

.qr-code{
  background-color: white;
  max-width: 60vw;
}

.SearchEvent{
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: center;
  justify-content: space-between;
  padding-right: 10px;
}

.selected-row {
  background-color: rgb(208, 238, 248);
  /* Change the background color to lightblue */
}

.ViewButton{
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  padding: 5px 10px;
  align-items: center;
  justify-content: center;
}

.ViewButtonLabel{
  margin: 0px;
  padding-left: 3px;
}