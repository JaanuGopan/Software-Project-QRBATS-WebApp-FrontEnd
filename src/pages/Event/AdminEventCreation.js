import "./EventCreation/EventCreation.css";
import eventCreationImage from "../../assets/Images/signin/Signin.jpeg";
import React, { useState, useRef } from "react";
import axios from "axios";
import html2canvas from "html2canvas";
import jsPDF from "jspdf";
import { IoMdCloseCircleOutline } from "react-icons/io";
import toast, { Toaster } from "react-hot-toast";
import { useSelector } from "react-redux";
import { selectUser } from "../../redux/features/userSlice";
import QRCode from "qrcode.react";
import SaveEventService from "../../api/services/SaveEventService";

const AdminEventCreation = ({
  handlecloseCreateEventWindow,
  reloadEventList,
}) => {
  const [eventId, setEventId] = useState("");
  const [eventName, setEventName] = useState("");
  const [moduleName, setModuleName] = useState(null);
  const [eventDate, setEventDate] = useState("");
  const [eventValidDate, setEventValidDate] = useState("");
  const [eventTime, setEventTime] = useState("");
  const [eventEndTime, setEventEndTime] = useState("");
  const [eventVenue, setEventVenue] = useState("");
  const [showQRCode, setShowQRCode] = useState(false);
  const [eventRole, setEventRole] = useState("EVENT");
  const [eventAssignedUserId, setEventAssignedUserId] = useState(null);

  const [showModuleNameInput, setShowModuleNameInput] = useState(true);
  const [title, setTitle] = useState("Event");

  const qrCodeRef = useRef(null);

  const venueList = [
    "NCC",
    "LT1",
    "LT2",
    "Auditorium",
    "DEIE",
    "DMME",
    "DCEE",
    "Other",
  ];

  const notifySuccess = () => toast.success("Successfully Event Created!");

  const clearEventDetails = () => {
    setEventName("");
    setEventDate("");
    setEventValidDate("");
    setEventTime("");
    setEventEndTime("");
    setEventVenue("");
    setEventAssignedUserId(null);
    setEventRole("EVENT");
  };

  const user = useSelector(selectUser);
  // Destructure user object for cleaner code
  const { userId } = user || {};

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      const response = SaveEventService.saveEvent(
        eventName,
        eventDate,
        eventValidDate,
        eventTime,
        eventEndTime,
        eventVenue,
        eventRole,
        moduleName,
        userId
      );
      setEventId(response.data.eventId);
      const responseEventName = response.data.eventName;
      notifySuccess();
      reloadEventList();
    } catch (error) {
      console.error("Event failed", error);
    } finally {
      setShowQRCode(true);
    }
  };

  const downloadQRCode = () => {
    const qrCodeURL = document
      .getElementById("qrCodeEl")
      .toDataURL("image/png")
      .replace("image/png", "image/octet-stream");
    let aEl = document.createElement("a");
    aEl.href = qrCodeURL;
    aEl.download = "QR_Code.png";
    document.body.appendChild(aEl);
    aEl.click();
    document.body.removeChild(aEl);
  };
  const eventDetails = {
    eventId: eventId,
    eventName: eventName,
    moduleName: moduleName,
    eventDate: eventDate,
    eventValidDate: eventValidDate,
    eventTime: eventTime,
    eventEndTime: eventEndTime,
    eventVenue: eventVenue,
    eventAssignedUserId: userId,
  };
  const qrCodeDetails = JSON.stringify(eventDetails);

  const handleShareQRCode = () => {
    html2canvas(qrCodeRef.current).then((canvas) => {
      const imgData = canvas.toDataURL("image/png");
      const pdf = new jsPDF();
      pdf.addImage(imgData, "PNG", 0, 0);
      pdf.save("qr_code.pdf");
    });
  };

  const [qrCodeWindow, setQrCodeWindow] = useState(false);

  return (
    <div className="event-main-container1">
      <Toaster />
      <div
        className="closeCreateEventWindow"
        onClick={handlecloseCreateEventWindow}
      >
        <IoMdCloseCircleOutline />
      </div>
      <h2>Create Event</h2>
      <div className="eventCreation-field">
        <img src={eventCreationImage} className="Create-logo" alt="Logo" />
        <div className="eventCreation-input-field">
          <form onSubmit={handleSubmit}>
            <div className="input-with-icon">
              <input
                required
                type="text"
                id="eventName"
                name="eventName"
                placeholder={"Event Name"}
                className="form-control mb-2"
                value={eventName}
                onChange={(e) => setEventName(e.target.value)}
              />
            </div>
            <div className="date-div">
              <div className="eventCreation-form">
                <div>
                  <label className="date-label" htmlFor="eventDate">
                    Event Starting Date
                  </label>
                  <input
                    required
                    type="date"
                    value={eventDate}
                    onChange={(e) => setEventDate(e.target.value)}
                    className="form-control mb-2"
                  />
                </div>
              </div>
              <div className="eventCreation-form">
                <div>
                  <label className="date-label" htmlFor="eventDate">
                    Event Ending Date
                  </label>
                  <input
                    required
                    type="date"
                    value={eventValidDate}
                    onChange={(e) => setEventValidDate(e.target.value)}
                    className="form-control mb-2"
                  />
                </div>
              </div>
            </div>
            <div className="eventCreation-form">
              <div>
                <label className="date-label" htmlFor="eventDate">
                  Venue
                </label>
                <select
                  required
                  value={eventVenue}
                  onChange={(e) => setEventVenue(e.target.value)}
                  className="form-control mb-2"
                >
                  <option value="">Select Venue</option>
                  {venueList.map((option, index) => (
                    <option key={index} value={option}>
                      {option}
                    </option>
                  ))}
                </select>
              </div>
            </div>
            <div className="eventCreation-form">
              <div>
                <label className="form-label" htmlFor="eventDate">
                  Event Starting Time
                </label>
                <input
                  required
                  type="time"
                  value={eventTime}
                  onChange={(e) => setEventTime(e.target.value)}
                  placeholder="Event Time"
                  className="form-control mb-2"
                />
              </div>
              <div className="eventCreation-form">
                <div>
                  <label className="form-label" htmlFor="enevtEndTime">
                    Event Ending Time
                  </label>
                  <input
                    required
                    type="time"
                    value={eventEndTime}
                    onChange={(e) => setEventEndTime(e.target.value)}
                    placeholder="Event End Time"
                    className="form-control mb-2"
                  />
                </div>
              </div>
            </div>

            <div className="eventCreation-form"></div>
            <button
              onClick={() => setQrCodeWindow(true)}
              type="submit"
              className="btn btn-primary w-100"
            >
              Create Event
            </button>
          </form>
        </div>
        {qrCodeWindow && showQRCode && (
          <div className="Admin-Create-Event-Dashboard">
            <div ref={qrCodeRef} className="event-main-container1">
              <div
                className="closeCreateEventWindow"
                onClick={handlecloseCreateEventWindow}
              >
                <IoMdCloseCircleOutline />
              </div>
              <h2>Successfully Event Created</h2>
              <div className="row-center">
                <QRCode
                  name="QRCode"
                  size={200}
                  id="qrCodeEl"
                  value={qrCodeDetails}
                  className="mb-2"
                />
              </div>
              <div>
                <p className="text-center">Scan this QR code to join {title}</p>
              </div>
              <div className="row-center">
                <div className="QRbutton">
                  <button
                    onClick={downloadQRCode}
                    className="btn btn-success mr-3"
                  >
                    Save
                  </button>
                </div>
              </div>
            </div>
          </div>
        )}
      </div>
    </div>
  );
};

export default AdminEventCreation;
