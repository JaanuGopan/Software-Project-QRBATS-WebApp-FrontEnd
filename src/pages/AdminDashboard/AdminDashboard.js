import React, { useState, useEffect } from "react";
import "./AdminDashboard.css";
import Table from "../../components/layout/AdminDashboardComponent/Table";
import TotalCount from "../../components/layout/AdminDashboardComponent/TotalCount";
import { FaUsers } from "react-icons/fa";
import { PiUsersFourFill } from "react-icons/pi";
import { FaSchool } from "react-icons/fa6";
import { IoNewspaperSharp } from "react-icons/io5";
import NormalButton from "../../components/layout/AdminDashboardComponent/NormalButton";
import { MdCreateNewFolder } from "react-icons/md";
import { RiDeleteBin5Fill } from "react-icons/ri";
import AdminUpdateEvent from "../Event/AdminUpdateEvent";
import AdminEventCreation from "../Event/AdminEventCreation";
import DeleteEventService from "../../api/services/DeleteEventService";
import FetchEventsService from "../../api/services/FetchEventsService";

const AdminDashboard = () => {
  const [eventCreatePopUpWindow, setEventCreatePopUpWindow] = useState(false);
  const [eventUpdatePopUpWindow, setEventUpdatePopUpWindow] = useState(false);
  const [selectedEvent, setSelectedEvent] = useState(null);
  const [eventList, setEventList] = useState([]);
  const [search, setSearch] = useState("");

  useEffect(() => {
    // Fetch the list of events from the API using the new class
    handleReloadEventList();
  }, []);

  // Function to handle event click
  const handleEventClick = (event) => {
    setSelectedEvent(event);
    // Do whatever you want with the selected event data
    console.log("Selected Event:", event);
  };

  const handleDelete = async () => {
    try {
      const response = await DeleteEventService.deleteEvent(
        selectedEvent.eventId
      );
      // After deleting, you may want to update the event list
      // Fetch the updated event list
      handleReloadEventList();
    } catch (error) {
      console.log("error " + error);
    }
  };

  const handleReloadEventList = async () => {
    FetchEventsService.fetchEvents()
      .then((events) => {
        setEventList(events);
      })
      .catch((error) => {
        console.error("Error fetching events:", error);
      });
  };

  return (
    <div className="admin-Dash">
      <p className="mainHead">Admin Dashboard</p>
      <div className="mainInform">
        <TotalCount
          total={"20"}
          countIcon={
            <FaUsers
              style={{ color: "white", padding: "2%", fontSize: "250%" }}
            />
          }
          countTitle={"Total Staffs"}
        />
        <TotalCount
          total={"201"}
          countIcon={
            <PiUsersFourFill
              style={{ color: "white", padding: "2%", fontSize: "250%" }}
            />
          }
          countTitle={"Total Students"}
        />
        <TotalCount
          total={"08"}
          countIcon={
            <FaSchool
              style={{ color: "white", padding: "2%", fontSize: "250%" }}
            />
          }
          countTitle={"Total Departments"}
        />
        <TotalCount
          total={"350"}
          countIcon={
            <IoNewspaperSharp
              style={{ color: "white", padding: "2%", fontSize: "250%" }}
            />
          }
          countTitle={"Total Modules"}
        />
      </div>
      <div className="SearchEvent">
        <p className="mainHead">Upcoming Events</p>
        <input
          type="text"
          placeholder="Search..."
          style={{
            width: "150px",
            padding: "3px 40px",
            border: "0.5px solid black",
            borderRadius: "5px",
            textAlign: "center",
          }}
          onChange={(e) => setSearch(e.target.value)}
        />
      </div>
      <div className="AdminEventList">
        <Table
          search={search}
          handleUpdateEvent={() => setEventUpdatePopUpWindow(true)}
          onEventClick={handleEventClick}
          eventList={eventList} // Pass the eventList prop here
        />

        <div className="List-Buttons">
          <NormalButton
            handleClick={() => setEventCreatePopUpWindow(true)}
            title={"Create"}
            titlewithiconicon={<MdCreateNewFolder className="buttonIcon" />}
          />
          <NormalButton
            title={"Delete"}
            handleClick={handleDelete}
            titlewithiconicon={<RiDeleteBin5Fill className="buttonIcon" />}
          />
        </div>
      </div>
      {eventCreatePopUpWindow && (
        <div
          handleClick={() => setEventCreatePopUpWindow(false)}
          className="Admin-Create-Event-Dashboard"
        >
          <AdminEventCreation
            handlecloseCreateEventWindow={() =>
              setEventCreatePopUpWindow(false)
            }
            reloadEventList={handleReloadEventList}
          />
        </div>
      )}
      {eventUpdatePopUpWindow && (
        <div
          handleClick={() => setEventCreatePopUpWindow(false)}
          className="Admin-Create-Event-Dashboard"
        >
          <AdminUpdateEvent
            handlecloseCreateEventWindow={() =>
              setEventUpdatePopUpWindow(false)
            }
            selectedEvent={selectedEvent}
            reloadEventList={handleReloadEventList}
          />
        </div>
      )}
    </div>
  );
};

export default AdminDashboard;
