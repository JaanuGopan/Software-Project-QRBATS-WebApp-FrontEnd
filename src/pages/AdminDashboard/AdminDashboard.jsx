import React, { useState, useEffect, useContext } from 'react';
import './AdminDashboard.css';
import Table from '../../components/layout/AdminDashboardComponent/Table';
import NormalButton from '../../components/layout/AdminDashboardComponent/NormalButton';
import { MdCreateNewFolder } from 'react-icons/md';
import { RiDeleteBin5Fill } from 'react-icons/ri';
import AdminUpdateEvent from '../Event/AdminUpdateEvent';
import AdminEventCreation from '../Event/AdminEventCreation';
import LocationService from '../../api/services/LocationService';
import EventService from '../../api/services/EventService';
import { toast, ToastContainer } from 'react-toastify';
import WarningPopup from '../../components/warningPopup/WarningPopup';
import { AuthContext } from '../../config/AuthProvider';

const AdminDashboard = () => {
  const [eventCreatePopUpWindow, setEventCreatePopUpWindow] = useState(false);
  const [eventUpdatePopUpWindow, setEventUpdatePopUpWindow] = useState(false);
  const [selectedEvent, setSelectedEvent] = useState(null);
  const [eventList, setEventList] = useState([]);
  const [search, setSearch] = useState('');
  const [venuesList, setVenuesList] = useState([]);

  const { user } = useContext(AuthContext);
  const { userId } = user || {};

  const handleGetLocationNameList = async () => {
    const response = await LocationService.getAllLocationNames();
    setVenuesList(response);
  };

  useEffect(() => {
    // Fetch the list of events from the API using the new class
    handleReloadEventList();
    handleGetLocationNameList();
  }, []);

  // Function to handle event click
  const handleEventClick = (event) => {
    setSelectedEvent(event);
    // Do whatever you want with the selected event data
    console.log('Selected Event:', event);
  };

  const [showDeleteEventPopup, setShowDeleteEventPopup] = useState(false);

  const showDeleteEventWarning = () => {
    if (selectedEvent === null) {
      toast.error('Please Select Event To Delete.');
      return;
    }
    setShowDeleteEventPopup(true);
  };

  const [processingDeleteEvent, setProcessingDeleteEvent] = useState(false);

  const handleDelete = async () => {
    try {
      setProcessingDeleteEvent(true);
      const response = await EventService.deleteEvent(selectedEvent.eventId);
      if (response.status === 200) {
        handleReloadEventList();
        setSelectedEvent(null);
        toast.success(`Successfully ${selectedEvent.eventName} Event Deleted.`);
      } else if (response.status === 400) {
        toast.error(response.data);
      } else {
        toast.error(`Error In Deleting Event ${selectedEvent.eventName}.`);
      }
    } catch (error) {
      console.log('error ' + error);
    } finally {
      setProcessingDeleteEvent(false);
      setShowDeleteEventPopup(false);
    }
  };

  const handleReloadEventList = async () => {
    EventService.getEventByUserID(userId)
      .then((events) => {
        setEventList(events);
      })
      .catch((error) => {
        console.error('Error fetching events:', error);
      });
  };

  return (
    <div className="admin-Dash">
      <ToastContainer />
      <p className="mainHead">Admin Dashboard</p>
      <div className="mainInform"></div>
      <div className="SearchEvent">
        <p className="mainHead">Upcoming Events</p>
        <input
          type="text"
          placeholder="Search..."
          style={{
            width: '150px',
            padding: '3px 40px',
            border: '0.5px solid black',
            borderRadius: '5px',
            textAlign: 'center',
          }}
          onChange={(e) => setSearch(e.target.value)}
        />
      </div>
      <div className="AdminEventList">
        <Table
          search={search}
          handleUpdateEvent={() => setEventUpdatePopUpWindow(true)}
          onEventClick={handleEventClick}
          eventList={eventList} // Pass the eventList prop here
        />

        <div className="List-Buttons">
          <NormalButton
            handleClick={() => setEventCreatePopUpWindow(true)}
            title={'Create'}
            titlewithiconicon={<MdCreateNewFolder className="buttonIcon" />}
            id={'Admincreate'}
          />
          <NormalButton
            title={'Delete'}
            handleClick={showDeleteEventWarning}
            titlewithiconicon={<RiDeleteBin5Fill className="buttonIcon" />}
          />
        </div>
      </div>
      {eventCreatePopUpWindow && (
        <div className="Admin-Create-Event-Dashboard">
          <AdminEventCreation
            handleCloseCreateEventWindow={() => setEventCreatePopUpWindow(false)}
            reloadEventList={handleReloadEventList}
            locationList={venuesList}
            showCloseButton={true}
          />
        </div>
      )}
      {eventUpdatePopUpWindow && (
        <div className="Admin-Create-Event-Dashboard">
          <AdminUpdateEvent
            handleCloseEventUpdateWindow={() => setEventUpdatePopUpWindow(false)}
            selectedEvent={selectedEvent}
            reloadEventList={handleReloadEventList}
            locationNameList={venuesList}
          />
        </div>
      )}

      {showDeleteEventPopup && (
        <div className="delete-event-container">
          <WarningPopup
            handleCloseWarningWindow={() => setShowDeleteEventPopup(false)}
            handleOk={handleDelete}
            titleText={'Are You Sure You Want To Delete This Event?'}
            buttonText={'Delete'}
            processing={processingDeleteEvent}
          />
        </div>
      )}
    </div>
  );
};

export default AdminDashboard;
