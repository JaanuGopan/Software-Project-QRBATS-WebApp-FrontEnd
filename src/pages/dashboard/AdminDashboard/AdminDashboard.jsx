import React, { useContext, useEffect, useState } from 'react';
import { FaPlus } from 'react-icons/fa';
import { toast } from 'react-toastify';
import EventService from '../../../api/services/EventService';
import LocationService from '../../../api/services/LocationService';
import AppContentCard from '../../../components/app-content-card/app-content-card';
import NormalButton from '../../../components/buttons/NormalButton';
import EventTable from '../../../components/tables/event-table';
import WarningPopup from '../../../components/warningPopup/WarningPopup';
import { AuthContext } from '../../../config/AuthProvider';
import AdminEventCreation from '../../Event/AdminEventCreation';
import AdminUpdateEvent from '../../Event/AdminUpdateEvent';
import './AdminDashboard.css';

const AdminDashboard = () => {
  const [eventUpdatePopUpWindow, setEventUpdatePopUpWindow] = useState(false);
  const [selectedEvent, setSelectedEvent] = useState(null);
  const [eventList, setEventList] = useState([]);
  const [search, setSearch] = useState('');
  const [venuesList, setVenuesList] = useState([]);
  const [showDeleteEventPopup, setShowDeleteEventPopup] = useState(false);
  const [processingDeleteEvent, setProcessingDeleteEvent] = useState(false);
  const [showEventCreationModal, setShowEventCreationModal] = useState(false);

  const { user } = useContext(AuthContext);
  const { userId } = user || {};

  const handleGetLocationNameList = async () => {
    const response = await LocationService.getAllLocationNames();
    setVenuesList(response);
  };

  useEffect(() => {
    handleReloadEventList();
    handleGetLocationNameList();
  }, []);

  const handleEventClick = (event) => {
    setSelectedEvent(event);
  };

  const showDeleteEventWarning = () => {
    setShowDeleteEventPopup(true);
  };

  const handleDelete = async () => {
    try {
      setProcessingDeleteEvent(true);
      const response = await EventService.deleteEvent(selectedEvent.eventId);
      if (response.status === 200) {
        handleReloadEventList();
        setSelectedEvent(null);
        toast.success(`Successfully ${selectedEvent.eventName} Event Deleted.`);
      } else if (response.status === 400) {
        toast.error(response.data);
      } else {
        toast.error(`Error In Deleting Event ${selectedEvent.eventName}.`);
      }
    } catch (error) {
      console.error('error ' + error);
    } finally {
      setProcessingDeleteEvent(false);
      setShowDeleteEventPopup(false);
    }
  };

  const handleReloadEventList = async () => {
    EventService.getEventByUserID(userId)
      .then((events) => {
        setEventList(events);
      })
      .catch((error) => {
        console.error('Error fetching events:', error);
      });
  };

  return (
    <div className="row">
      <div className="col py-2">
        <div className="row my-1 align-items-center justify-content-between">
          <div className="col-auto ms-4 fs-3 fw-bold ">Upcoming Events</div>
          <div className="col-auto">
            <div className="row align-items-center">
              <div className="col-auto">
                <NormalButton
                  title={'Create Event'}
                  handleClick={() => setShowEventCreationModal(true)}
                  icon={<FaPlus className="buttonIcon" />}
                />
              </div>
              <div className="col-auto">
                <input
                  type="text"
                  placeholder="Search..."
                  style={{
                    width: '150px',
                    padding: '3px 40px',
                    border: '0.5px solid black',
                    borderRadius: '5px',
                    textAlign: 'center',
                  }}
                  onChange={(e) => setSearch(e.target.value)}
                />
              </div>
            </div>
          </div>
        </div>
        <AppContentCard>
          <EventTable
            search={search}
            handleUpdateEvent={() => setEventUpdatePopUpWindow(true)}
            onEventClick={handleEventClick}
            eventList={eventList}
            handleEventDelete={showDeleteEventWarning}
          />
        </AppContentCard>
        {showEventCreationModal && (
          <div className="Admin-Create-Event-Dashboard">
            <AdminEventCreation
              handleCloseCreateEventWindow={() => setShowEventCreationModal(false)}
              reloadEventList={handleReloadEventList}
              locationList={venuesList}
              showCloseButton={true}
            />
          </div>
        )}
        {eventUpdatePopUpWindow && (
          <div className="Admin-Create-Event-Dashboard">
            <AdminUpdateEvent
              handleCloseEventUpdateWindow={() => setEventUpdatePopUpWindow(false)}
              selectedEvent={selectedEvent}
              reloadEventList={handleReloadEventList}
              locationNameList={venuesList}
            />
          </div>
        )}
        {showDeleteEventPopup && (
          <div className="delete-event-container">
            <WarningPopup
              handleCloseWarningWindow={() => setShowDeleteEventPopup(false)}
              handleOk={handleDelete}
              titleText={'Are You Sure You Want To Delete This Event?'}
              buttonText={'Delete'}
              processing={processingDeleteEvent}
            />
          </div>
        )}
      </div>
    </div>
  );
};

export default AdminDashboard;
