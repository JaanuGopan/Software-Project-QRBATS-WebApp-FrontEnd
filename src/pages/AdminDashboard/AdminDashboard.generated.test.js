import renderer from 'react-test-renderer';
import React, { useState, useEffect } from "react";

import Table from "../../components/layout/AdminDashboardComponent/Table";
import TotalCount from "../../components/layout/AdminDashboardComponent/TotalCount";
import { FaUsers } from "react-icons/fa";
import { PiUsersFourFill } from "react-icons/pi";
import { FaSchool } from "react-icons/fa6";
import { IoNewspaperSharp } from "react-icons/io5";
import NormalButton from "../../components/layout/AdminDashboardComponent/NormalButton";
import { MdCreateNewFolder } from "react-icons/md";
import { RiDeleteBin5Fill } from "react-icons/ri";
import AdminUpdateEvent from "../Event/AdminUpdateEvent";
import AdminEventCreation from "../Event/AdminEventCreation";
import DeleteEventService from "../../api/services/DeleteEventService";
import FetchEventsService from "../../api/services/FetchEventsService";
import AdminDashboard from "./AdminDashboard";

jest.mock("./AdminDashboard.css");
jest.mock("../../components/layout/AdminDashboardComponent/Table");
jest.mock("../../components/layout/AdminDashboardComponent/TotalCount");
jest.mock("react-icons/fa");
jest.mock("react-icons/pi");
jest.mock("react-icons/fa6");
jest.mock("react-icons/io5");
jest.mock("../../components/layout/AdminDashboardComponent/NormalButton");
jest.mock("react-icons/md");
jest.mock("react-icons/ri");
jest.mock("../Event/AdminUpdateEvent");
jest.mock("../Event/AdminEventCreation");
jest.mock("../../api/services/DeleteEventService");
jest.mock("../../api/services/FetchEventsService");

const renderTree = tree => renderer.create(tree);
describe('<AdminDashboard>', () => {
  it('should render component', () => {
    expect(renderTree(<AdminDashboard 
    />).toJSON()).toMatchSnapshot();
  });
  
});