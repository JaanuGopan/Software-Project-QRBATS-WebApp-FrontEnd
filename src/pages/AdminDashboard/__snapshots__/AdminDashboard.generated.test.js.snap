// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminDashboard> should render component 1`] = `
<div
  className="admin-Dash"
>
  <p
    className="mainHead"
  >
    Admin Dashboard
  </p>
  <div
    className="mainInform"
  />
  <div
    className="SearchEvent"
  >
    <p
      className="mainHead"
    >
      Upcoming Events
    </p>
    <input
      onChange={[Function]}
      placeholder="Search..."
      style={
        Object {
          "border": "0.5px solid black",
          "borderRadius": "5px",
          "padding": "3px 40px",
          "textAlign": "center",
          "width": "150px",
        }
      }
      type="text"
    />
  </div>
  <div
    className="AdminEventList"
  >
    <div
      className="List-Buttons"
    />
  </div>
</div>
`;
